debian/control

```diff
@@ -9,7 +9,7 @@ Build-Depends: bash-completion,
                dh-systemd | debhelper (>= 13.3),
                gettext,
                git,
-               golang,
+               golang-1.14 [!powerpc] | golang-1.10 [!powerpc !riscv64],
```

Comment:
This is why this _should_ work (lmk if you think this should be documented outside a github comment)
* go-1.14 is on focal and later
* go-1.14 supports risc v, and risc v ubuntu is supported on focal and later
* go-1.14 doesn't support powerpc but neither does focal or after
* go-1.10 is on bionic and earlier
* go-1.10 doesn't support risc v but neither does bionic or earlier
* go-1.10 doesn't support powerpc, so we wont deliver the hook on any powerpc bionic or earlier

So the scenarios are:
1. Any release on powerpc - nothing is satisfied, golang should be ignored and we won't deliver json-hook
2. Bionic or earlier on non-powerpc - golang-1.14 won't exist so use golang-1.10
3. Focal or later on anything (including riscv) - use first option golang-1.14

To test manually:
1. Simulate (1) above by replacing "powerpc" in this line with "amd64" and then testing a build on bionic or earlier
    - Expect a successful build and install. Verify that the jsonhook was not built and not installed
2. Simulate (2) above by using this line as-is, and testing a build on bionic or earlier
    - Expect a successful build and install. Verify that the jsonhook was built with 1.10 and installed
3. Simulate (3) above by replacing "riscv64" in this line with "amd64" and then testing a build on focal or later
    - Expect a successful build and install. Verify that the jsonhook was built with 1.14 and installed

Future issues:
* When we stop packaging go-1.14 for the latest release, we'll need to add another `|` alternative to this line.

Question: What is the best way to test this "for real" on actual/virtual powerpc/riscv machines?

Reply:
I don't think we should depend on fixed versions of packages, mainly for the future issue you also noted, but also for correctness (we don't need that *specific* version). What about Build-Depending on:
```
golang-go (>= 2:1.10~) | golang-1.13-go <!nocheck>, 
```
This should ensure that a recent enough version of `golang-go` gets installed, falling back to `golang-1.13-go` on Xenial, where `golang-go` would depend on the old `golang-1.6-go`. I hope it makes sense.

The `<!nocheck>` skips installing the dependency if the the test stage of the package build process is to be skipped. I think this is correct as it's actually a test dependency here (note the `nocheck` in the `override_dh_auto_test` target in `d/rules`).

Note that the arch is called `ppc64el`, while `powerpc` is the legacy one of the old Macs (long deprecated, see https://help.ubuntu.com/community/SupportedArchitectures#PowerPC_.28legacy.29).

Reply:
@paride Thanks for this feedback! This is helpful. I like your suggestion and I agree we should have `golang-go` as the first choice rather than a versioned package. I also think that its a good idea to use the `golang-go` packages instead of the whole `golang` packages. However, this suggestion won't quite satisfy all of our requirements. I'll list the requirements here as I understand them.

* To build on most architectures, we need >=1.4 (or so, I haven't actually tested the lower bound here).
* To test on any arch, we need >=1.7
* To build for `riscv64` we need >= 1.14 (and riscv is on focal and newer)
* We need to gracefully handle that no version of go supports `powerpc`
    * My understanding was that the issue is with the legacy `powerpc` builds failing for xenial, not the newer `ppc64el`, which I think should be fine. **I will double check this though.**

So I think with `golang-go (>= 2:1.10~) | golang-1.13-go`, we'd end up with `golang-go=2:1.13~1ubuntu2` on focal which would fail to build `riscv64`, and we would still fail the `powerpc` builds.

But based on your suggestion, I think the following would work:
```
golang-go (>= 2:1.14~) [!powerpc] | golang-1.14-go [!powerpc] | golang-1.13-go [!powerpc !riscv64],
```
So we'd get `golang-go` on groovy and newer, we'd get `golang-1.14-go` on focal, and we'd get `golang-1.13-go` on xenial. We'd completely ignore the dependency for `powerpc` (still going to double check that). And it explicitly states that 1.13 won't support riscv, though I'm not sure if that bit is necessary.

What do you think?